src/pluggable_device.rs

```diff
@@ -0,0 +1,53 @@
+use crate::{Result, Status};
+use std::ffi::CString;
+use tensorflow_sys as tf;
+
+/// PluggableDeviceLibrary handler.
+#[derive(Debug)]
+pub struct PluggableDeviceLibrary {
+    inner: *mut tf::TF_Library,
+}
+
+impl PluggableDeviceLibrary {
+    /// Load the library specified by library_filename and register the pluggable
+    /// device and related kernels present in that library. This function is not
+    /// supported on embedded on mobile and embedded platforms and will fail if
+    /// called.
+    ///
+    /// Pass "library_filename" to a platform-specific mechanism for dynamically
+    /// loading a library. The rules for determining the exact location of the
+    /// library are platform-specific and are not documented here.
+    pub fn load(library_filename: &str) -> Result<PluggableDeviceLibrary> {
+        let status = Status::new();
+        let library_filename = CString::new(library_filename)?;
+        let lib_handle =
+            unsafe { tf::TF_LoadPluggableDeviceLibrary(library_filename.as_ptr(), status.inner) };
+        status.into_result()?;
+
+        Ok(PluggableDeviceLibrary { inner: lib_handle })
+    }
+}
+
+impl Drop for PluggableDeviceLibrary {
+    /// Frees the memory associated with the library handle.
+    /// Does NOT unload the library.
+    fn drop(&mut self) {
+        unsafe {
+            tf::TF_DeletePluggableDeviceLibraryHandle(self.inner);
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[ignore]
+    #[test]
+    fn load_pluggable_device_library() {
```

Comment:
We might as well leave the test out, since there's no reasonable way to run it (as far as I know).  We have similar issues with TF_LoadLibrary.

Reply:
Yes, right. I tried some, but none of them succeeded. There is too little information on this API.
 
Is there anything else I can do with this pull request?

Reply:
In https://github.com/tensorflow/rust/pull/387 they solved it (at least for macOS) by installing tensorflow-metal and loading that.  If you want to leave this test out, I can merge this first and ask them to rebase their changes on top of this, so we'd at least have a test that runs in the CI, even if not on everyone's local machine.

Reply:
Thanks for adjusting the PRs. If that order is acceptable, please do so.

A while ago, I was trying to see if the C-API would work with a plugin for Windows, and latest plugin version (currently build from source only) seemed to work with TF 2.12. I'm thinking of trying to see if I can support that as well (apart from this PR) when the version of tensorflow here is updated.

https://github.com/microsoft/tensorflow-directml-plugin